pkg/controller/enterprisesearch/enterprisesearch_controller.go

```diff
@@ -290,6 +285,37 @@ func (r *ReconcileEnterpriseSearch) validate(ctx context.Context, ent *entv1beta
 	return nil
 }
 
+func (r *ReconcileEnterpriseSearch) updateStatus(ent entv1beta1.EnterpriseSearch, deploy appsv1.Deployment, svcName string) error {
+	newStatus := entv1beta1.EnterpriseSearchStatus{
+		Health: entv1beta1.EnterpriseSearchRed,
+		ReconcilerStatus: commonv1.ReconcilerStatus{
+			AvailableNodes: deploy.Status.AvailableReplicas,
+		},
+		ExternalService: svcName,
+		Association:     ent.Status.Association,
+	}
+	for _, c := range deploy.Status.Conditions {
+		if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+			newStatus.Health = entv1beta1.EnterpriseSearchGreen
+		}
+	}
+
+	if reflect.DeepEqual(newStatus, ent.Status) {
+		return nil // nothing to do
```

Comment:
I was wondering if this branch does not deserve a test since it protects the API Server from unnecessary calls.

Reply:
You're right. Do you have an idea for how we could test this? (mocking apiserver calls etc. feels a bit overkill)

Reply:
May be we can add a wrapper around the `Status()` receiver ?

```golang
type fakeClient struct {
	updateCalled bool
	k8s.Client
}

func newFakeClient(initObjs ...runtime.Object) k8s.Client {
	return &fakeClient{
		false,
		k8s.WrappedFakeClient(initObjs...),
	}

}

func (f *fakeClient) Status() k8s.StatusWriter {
	f.updateCalled = true // Status() has been requested for update
	return f.Client.Status()
}
```

Reply:
Yeah let's do that!